torchtune/utils/ppo_utils/collate.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Dict, List
+
+import torch
+from torch.nn.utils.rnn import pad_sequence
+
+
+def left_padded_collate(
```

Comment:
Personally I wonder whether this is a sufficiently general function to just add to `torchtune/utils/collate.py`. The one thing I could see being a bit confusing is that the return signatures differ from our existing `padded_collate`

Reply:
Yeah there's already a padded_collate_dpo, we should add this to our collator functions, but also make the signature match` e.g. Dict[str, Tensor]`

Reply:
I'm happy to make this general if you think it's of use elsewhere. It's mainly like this because a) I was being lazy and didn't want to handle logic for padding label/input IDs of differing lengths, b) we only ever want prompt inputs for PPO where this was being used. 
I could move the core logic from this function to `left_pad` and add a `ppo_collate` which only applies left-padding to the input tokens. If someone wanted to use this kind of padding they could then re-use the underlying logic by defining their own collate function.